sceptre/cli/__init__.py

```diff
@@ -63,7 +63,17 @@ def cli(
     }
     user_variables = {}
     if var_file:
-        user_variables.update(yaml.safe_load(var_file.read()))
+        for fh in var_file:
+            parsed = yaml.safe_load(fh.read())
+            # intersection
+            overloaded_keys = set(user_variables.keys()) & set(parsed.keys())
+            if overloaded_keys:
+                logger.debug(
+                    "Duplicate variables encountered: {0}. "
+                    "Using values from file with highest precedence: '{1}'."
+                    .format(', '.join(overloaded_keys), fh.name)
+                )
+            user_variables.update(parsed)
     if var:
         # --var options overwrite --var-file options
```

Comment:
and we should also probably put a debug with similar effect here as well

Reply:
done.